docs/output.md

```diff
@@ -258,6 +258,91 @@ print(result.output)
 
 _(This example is complete, it can be run "as is")_
 
+#### Handling partial output
+
+When streaming with `run_stream()`, output validators and output functions are called **multiple times** — once for each partial output received from the model, and once for the final complete output.
+
+For output validators or functions with **side effects** (e.g., sending notifications, logging, database updates), you should check the [`RunContext.partial_output`][pydantic_ai.tools.RunContext.partial_output] flag to avoid executing side effects on partial data.
```

Comment:
We should also mention you can use this when only the complete result should be validated, not intermediate values